### PR TITLE
refactor: move around some helpers before state-sync

### PR DESCRIPTION
**Pull request by mhofman**
refs: #3769

## Description

This extracts a few pure refactor commits from https://github.com/Agoric/agoric-sdk/pull/7225 in an attempt to make that PR more digestible.

### Security Considerations

None

### Scaling Considerations

None

### Documentation Considerations

None

### Testing Considerations

None

### CODE REVIEW

**Comment by mhofman** [comment]
Sorry for the temporary churn, I mistakenly cherry-picked the wrong commit and pressed push too quickly.

File: packages/internal/src/shutdown.js
Comment (michaelfig) [inline packages/internal/src/shutdown.js:1]: I'm not sure what to think about this.  Part of me agrees that we can have platform dependencies in `@agoric/internal` if we don't export those modules from `'@agoric/internal'`, rather by using a deep import instead.  But in practice, those platform dependencies are awfully hard to diagnose if an endo-bundled module accidentally imports one by contagion.

Could you instead move the Node.js dependents to `internal/src/node/*`, so at least it is really obvious when we import them or decide to find them a new home?

Comment (mhofman) [reply packages/internal/src/shutdown.js:1]: Done for `shutdown` and other helpers relying on `node`. PTAL

**Review by michaelfig — APPROVED** [review]
LGTM!  Onward and upward....

**Comment by mhofman** [comment]
@mergifyio requeue

**Pull request merged**